### PR TITLE
Rpc networking

### PR DESCRIPTION
**Pull request by paulfitz**
Just to add to the stack of conflicting PRs :-).  This is a sketch of how I'd like to be able to use Rpc.  I'd like to be able to name an entity in one place, and communicate with it from elsewhere, without being required to know how to get from one to the other.

I feel like the solution here ends up with more kinds of names than needed, but hopefully gets the idea across.

### CODE REVIEW

File: test/test_node.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const node5 = new Node("node5");
+    connect(1.0, 50, node1, node2);
+    connect(1.0, 50, node2, node3);
+    connect(0.0, 0, node3, node4);
+    connect(0.0, 0, node3, node5);
+    connect(10.0, 500, node1, node5);
+
+    const rpc1 = new Rpc();
+    rpc1.registerImpl<ICalc>("calc", new Calc());
+    node1.listen(rpc1, 'backend');
```
Comment (cpind) [inline test/test_node.ts]: Does it mean plugin author is exposed a `node` object corresponding to the plugin process ? 
I guess, I don't really see what lines plugin author has to write in order to call/register function across frontend / server.

Comment (paulfitz) [reply test/test_node.ts]: I don't really understand your question @cpind.

The `node1.listen`/`node1.connect` lines use global names to set up a pipe between rpc instances.  After that, all communication between those two rpcs is as before.

**Comment by dsagal** [comment]
To summarize our conversation over skype, I'd be satisfied with either of the proposals for routing, since they seem to offer a fairly similar interface to Rpc users. All of them would allow simplifying some code currently in Grist. I'd say whoever does the simplifying of the code in Grist gets to pick which proposal they like best :)

**Comment by paulfitz** [comment]
To summarize where all this lead: a forwarding implementation was merged https://github.com/gristlabs/grain-rpc/pull/5

**Comment by paulfitz** [comment]
Closing, outdated.

**Pull request closed without merging**